atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2123,6 +2124,65 @@ public void compactInternally(TableReference tableRef) {
         }
     }
 
+    @Override
+    public NodeAvailabilityStatus getNodeAvailabilityStatus() {
+        if (!doesConfigReplicationFactorMatchWithCluster()) {
+            return NodeAvailabilityStatus.TERMINAL;
+        }
+        return getStatusByRunningOperationsOnEachHost();
```

Comment:
My guess is we're best off either reversing the order here or doing all the actual terminal checks after we've confirmed a node is healthy like how the partitioner check is currently being ran.